modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/render/DateDDMFormFieldValueRenderer.java

```diff
@@ -64,9 +64,15 @@ private String _format(Serializable value, Locale locale) {
 
 			String pattern = simpleDateFormat.toPattern();
 
+			String yearPattern = "\\byy\\b";
+
+			if (pattern.endsWith("/y")) {
```

Comment:
@MrJohn1911 This fixes the unit test because we only test Locale.US(MM/dd/yy) and Locale.Brazil(dd/MM/y). but what about other locales that may not end with "/y"? I think those need to be converted to yyyy as well.

if you check this line with jdk21 and CLDR, you can see that many locales now have single "y"
`Arrays.stream(Locale.getAvailableLocales()).map(locale -> locale.toLanguageTag() + ": " + ((SimpleDateFormat)DateFormat.getDateInstance(SimpleDateFormat.SHORT, locale)).pattern).filter(str->str.matches(".*\\by\\b.*")).collect(
	Collectors.toList())`
![image](https://github.com/liferay-core-infra/liferay-portal/assets/71372051/de1b49c3-fa8a-423d-a5eb-df2a5ab3742f)

I think you might just need to do this
```
pattern = pattern.replaceAll("\\byy\\b", "yyyy");
pattern = pattern.replaceAll("\\by\\b", "yyyy");
```